app/src/main/java/com/babylon/wallet/android/presentation/settings/troubleshooting/importlegacywallet/ImportLegacyWalletViewModel.kt

```diff
@@ -439,14 +438,10 @@ class ImportLegacyWalletViewModel @Inject constructor(
 
     fun onContinueWithLedgerClick() {
         viewModelScope.launch {
-            val hasAtLeastOneLinkedConnector = getProfileUseCase().appPreferences.p2pLinks().isNotEmpty()
+            val hasAtLeastOneLinkedConnector = getProfileUseCase().appPreferences.p2pLinks.isNotEmpty()
 
             if (hasAtLeastOneLinkedConnector) {
                 useLedgerDelegate.onSendAddLedgerRequest()
-            } else if (hasAtLeastOneLinkedConnector.not()) {
-                _state.update {
-                    it.copy(shouldShowAddLinkConnectorScreen = true)
-                }
```

Comment:
Please take a look at this. Else block would never be executed. So I had to change this method. Please make sure you read this and tell me if the change is ok.

Reply:
This changes the current behavior by also updating `addLedgerSheetState`, unless it already has the same value: `AddLedgerDeviceUiState.ShowContent.AddLedgerDeviceInfo`

Reply:
@giannis-rdx Changed it.

Reply:
if there is at least one p2plink in the profile then send the request,
otherwise show the "add link connector "screen.
We don't need the "add ledger device" screen in this particular flow.